rest/data/host.go

```diff
@@ -80,6 +81,26 @@ func (hc *DBHostConnector) NewIntentHost(distroID, keyNameOrVal, taskID, userDat
 	return intentHost, nil
 }
 
+func (hc *DBHostConnector) SetHostStatus(host *host.Host, status string) error {
+	return host.SetStatus(status)
+}
+
+func (hc *DBHostConnector) SetHostPassword(ctx context.Context, host *host.Host, password string) error {
+	return spawn.SetHostRDPPassword(ctx, host, password)
```

Comment:
I'm a little uneasey about adding the first context into the host connector API. I don't think it's wrong, but I think its probably larger change than it appears. 

Additionally, I think the spawn method that you've extracted might be more reasonable to just call directly, unless you need/want to mock it out (and we might be able to think of better ways than just using the DB connector...